extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/KeycloakPolicyEnforcerConfig.java

```diff
@@ -0,0 +1,192 @@
+package io.quarkus.keycloak.pep;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.keycloak.representations.adapters.config.PolicyEnforcerConfig;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.runtime.annotations.DefaultConverter;
+
+@ConfigRoot(name = "keycloak", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class KeycloakPolicyEnforcerConfig {
+
+    /**
+     * Adapters will make separate HTTP invocations to the Keycloak server to turn an access code into an access token.
+     * This config option defines how many connections to the Keycloak server should be pooled
+     */
+    @ConfigItem(defaultValue = "20")
+    int connectionPoolSize;
+
+    /**
+     * Policy enforcement configuration when using Keycloak Authorization Services
+     */
+    @ConfigItem
+    KeycloakConfigPolicyEnforcer policyEnforcer;
+
+    @ConfigGroup
+    public static class KeycloakConfigPolicyEnforcer {
+
+        /**
+         * Specifies how policies are enforced.
+         */
+        @ConfigItem
+        boolean enable;
+
+        /**
+         * Specifies how policies are enforced.
+         */
+        @ConfigItem(defaultValue = "ENFORCING")
+        String enforcementMode;
+
+        /**
+         * Specifies the paths to protect.
+         */
+        @ConfigItem
+        Map<String, PathConfig> paths;
+
+        /**
+         * Defines how the policy enforcer should track associations between paths in your application and resources defined in
+         * Keycloak.
+         * The cache is needed to avoid unnecessary requests to a Keycloak server by caching associations between paths and
+         * protected resources
+         */
+        @ConfigItem
+        PathCacheConfig pathCache;
+
+        /**
+         * Specifies how the adapter should fetch the server for resources associated with paths in your application. If true,
+         * the
+         * policy
+         * enforcer is going to fetch resources on-demand accordingly with the path being requested
+         */
+        @ConfigItem(defaultValue = "true")
+        boolean lazyLoadPaths;
+
+        /**
+         * Defines a URL where a client request is redirected when an "access denied" message is obtained from the server.
+         * By default, the adapter responds with a 403 HTTP status code
+         */
+        @ConfigItem
+        Optional<String> onDenyRedirectTo;
```

Comment:
I believe this has to be managed by the `quarkus-oidc` - this is exactly what it will already do as part of your authorization code flow :-) or the existing bearer-only check. I mean it will do 401 if the token is invalid (signature for ex), but if the token has not enough permissions, it is equally invalid. And we let the user be redirected to something local then we'll lose the flow control, what will happen after the user lands on some local page, as opposed to be redirected to IDP ?

Reply:
This is not really related to code flow. This option, in particular, is just providing an option to customize requests denied by the server. The user is already authenticated and if not, this page is not meant to start any flow.

For instance, instead of returning a 403 response the application may want to show a more fancy page.

Reply:
@pedroigor It looks like the error can only occur in the process of the user authenticating (for the bearer only flow it is only 403 anyway). So if it is SPA then it is only really an adapter level issue what happens to a user view. And if it is a server web app code flow then the user has signed in at the IDP site, is being redirected now to the adapter, the token fails a check, and with custom page the user lands there seeing that the access is denied, But what is next ? The user stays authenticated ? Does not seem like the right idea.
The other thing, I doubt that in production people will want to display the details outside of its log :-)

Perhaps we should have an option for OIDC adapter itself return a page like "Your access to this resource has been denied, please contact the administrator, you will be redirected to IDO in 5 sec..." (and this page can be customized). 

But whatever we do here, it feels like it is not a KC specific extension concern since 403 will be possible at the `quarkus-oidc` level as well :-)

Reply:
That is the idea. This requirement came from the community a long time ago.

The user would just see this custom page with whatever the application wants to show. Either a simple and fancy "Access Denied" or something else.

If we want to have that on `quarkus-oidc`, that is fine and I can remove this.

Reply:
@pedroigor yes please lets remove if for the moment and then figure out where to have this property - the same concept should work across the OIDC bits in Quarkus